pkg/controller/monitor/monitor_controller.go

```diff
@@ -262,10 +292,19 @@ func (r *ReconcileMonitor) Reconcile(ctx context.Context, request reconcile.Requ
 		r.status.SetDegraded(operatorv1.ResourceCreateError, "Unable to create the Tigera CA", err, reqLogger)
 		return reconcile.Result{}, err
 	}
-	serverTLSSecret, err := certificateManager.GetOrCreateKeyPair(r.client, monitor.PrometheusServerTLSSecretName, common.OperatorNamespace(), PrometheusTLSServerDNSNames(r.clusterDomain))
-	if err != nil {
-		r.status.SetDegraded(operatorv1.ResourceCreateError, "Error creating TLS certificate", err, reqLogger)
-		return reconcile.Result{}, err
+
+	var serverTLSSecret certificatemanagement.KeyPairInterface
+	if !externalPrometheus || install.CertificateManagement != nil {
```

Comment:
I think this branch could use a comment.

```
// We're either not using an external prometheus in which case we simply sign the KeyPair ourselves, or we're configured to use an external CSR signer to provision certificates which is also handled under the covers by `GetOrCreateKeyPair`.
```

I can see the distinction between `CreateCSRKeyPair` and the CertificateManagement code that is implemented in `GetOrCreateKeyPair` (that also uses CSRs ultimately) to be pretty confusing.